psydac/linalg/utilities.py

```diff
@@ -198,3 +201,58 @@ def _sym_ortho(a, b):
         s = c * tau
         r = a / c
     return c, s, r
+
+#==============================================================================
+class SparseMatrixLinearOperator(MatrixFreeLinearOperator):
```

Comment:
Why do you need to subclass `MatrixFreeLinearOperator`?

`MatrixFreeLinearOperator` is a class designed for creating a `LinearOperator` object with a user-defined `dot` method provided to the constructor. In general I do not see much use for it. Unless you find it convenient to use its `__init__` method?

In this case you are creating a new class, so you could simply subclass `LinearOperator` and implement the required properties `domain`, `codomain`, and `dtype`, and methods `tosparse`, toarray`, `dot`, and `transpose`.

Reply:
I used it in the beginning as convenience, because I thought it'll be simpler. But when doing the latest changes, I asked myself the same question. 

I'll make it a subclass of `LinearOperator` and move it to its own file.